### DISCUSSION TITLE
Error build kbengine 1.3.13 in ubuntu 20.04

### DISCUSSION THREAD

**Discussion by yuriandradesp** [kbengine/kbengine#1435] [Q&A]
Hello guys,

Help me:

make[2]: Leaving directory '/home/kbe/kbengine/kbe/src/lib/network'
cd helper && make all
make[2]: Entering directory '/home/kbe/kbengine/kbe/src/lib/helper'
g++ -m64 -fPIC -pipe -Wall -Wno-deprecated -Wno-uninitialized -Wno-char-subscripts -fno-strict-aliasing -Wno-non-virtual-dtor -Wno-invalid-offsetof -Werror -std=c++11 -O3 -g -DLOG4CXX_STATIC -DENABLE_WATCHERS -DUSE_SIGAR -DUSE_JEMALLOC -DKBE_SERVER -MMD -DKBE_CONFIG="Hybrid64" -DCODE_INLINE -DKBE_USE_ASSERTS -D_HYBRID -I /home/kbe/kbengine/kbe/src -I /home/kbe/kbengine/kbe/src/lib -I /home/kbe/kbengine/kbe/src/server -I /home/kbe/kbengine/kbe/src/lib/dependencies -I /home/kbe/kbengine/kbe/src/lib/dependencies/zlib -I /home/kbe/kbengine/kbe/src/lib/dependencies/tinyxml -I /home/kbe/kbengine/kbe/src/lib/dependencies/fmt/include -I /home/kbe/kbengine/kbe/src/lib/dependencies/curl/include -I /home/kbe/kbengine/kbe/src/lib/dependencies/log4cxx/src/main/include -I/usr/include -I/home/kbe/kbengine/kbe/src/lib/dependencies/g3dlite -I/home/kbe/kbengine/kbe/src/lib/dependencies/sigar/linux -I/home/kbe/kbengine/kbe/src/lib/dependencies/jwsmtp/jwsmtp/jwsmtp -I/home/kbe/kbengine/kbe/src/lib/dependencies/tmxparser -I/home/kbe/kbengine/kbe/src/lib/dependencies/jemalloc/include -c debug_helper.cpp -o Hybrid64/debug_helper.o
debug_helper.cpp: In member function ‘void KBEngine::DebugHelper::onMessage(KBEngine::uint32, const char*, KBEngine::uint32)’:
debug_helper.cpp:707:12: error: ‘int ftime(timeb*)’ is deprecated [-Werror=deprecated-declarations]
707 | ftime(&tp);
| ^
In file included from debug_helper.cpp:40:
/usr/include/x86_64-linux-gnu/sys/timeb.h:39:12: note: declared here
39 | extern int ftime (struct timeb __timebuf)
| ^~~~~
debug_helper.cpp:707:12: error: ‘int ftime(timeb)’ is deprecated [-Werror=deprecated-declarations]
707 | ftime(&tp);
| ^
In file included from debug_helper.cpp:40:
/usr/include/x86_64-linux-gnu/sys/timeb.h:39:12: note: declared here
39 | extern int ftime (struct timeb __timebuf)
| ^~~~~
debug_helper.cpp:755:12: error: ‘int ftime(timeb)’ is deprecated [-Werror=deprecated-declarations]
755 | ftime(&tp);
| ^
In file included from debug_helper.cpp:40:
/usr/include/x86_64-linux-gnu/sys/timeb.h:39:12: note: declared here
39 | extern int ftime (struct timeb __timebuf)
| ^~~~~
debug_helper.cpp:755:12: error: ‘int ftime(timeb)’ is deprecated [-Werror=deprecated-declarations]
755 | ftime(&tp);
| ^
In file included from debug_helper.cpp:40:
/usr/include/x86_64-linux-gnu/sys/timeb.h:39:12: note: declared here
39 | extern int ftime (struct timeb *__timebuf)
| ^~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [/home/kbe/kbengine/kbe/src/build/common.mak:525: Hybrid64/debug_helper.o] Error 1
make[2]: Leaving directory '/home/kbe/kbengine/kbe/src/lib/helper'
make[1]: *** [Makefile:60: all] Error 2
make[1]: Leaving directory '/home/kbe/kbengine/kbe/src/lib'
make: *** [Makefile:7: all] Error 2

**Comment by ghost** [comment]
Try to remove Werror flag. All the errors are actually warnings.

**Reply by yuriandradesp** [reply]
Thanks for the answer. But how do I remove it?

**Reply by kbengine** [reply]
SYNOPSIS
       #include <sys/timeb.h>

       int ftime(struct timeb *tp);

DESCRIPTION
       NOTE: This function is deprecated, and will be removed in a future version of the GNU C library. Use clock_gettime(2) instead.

**Reply by huhubis** [reply]
Update to the latest
